### ISSUE TITLE
Support updating configmap

### ISSUE THREAD

**Issue by msau42** [kubernetes-sigs/sig-storage-local-static-provisioner#12]
Migrating from https://github.com/kubernetes-incubator/external-storage/issues/1011
/kind feature

**Comment by cofyc** [comment]
I was asked several times by users why they applied new yaml files but provisioner cannot discover for new storage classes.
If no one is looking at it, I'll take it.
/assign

**Comment by bcollard** [comment]
Hello, 

What about guiding people to use Kustomize + patches or configMapGenerator, so that for each update of the CM, the DaemonSet is notified and updated with a new CM name.
This would need a step-by-step guide.

**Comment by bcollard** [comment]
Please, what's your opinion about this?
My feeling is that most people need to deploy new apps regularly. So we cannot offer a static configuration through this configMap / DaemonSet combo.

What's the proposed way of working when one want to deploy a new app requiring some PV(s)?
How to update the DS and CM properly?

**Comment by cofyc** [comment]
you need to update the ConfigMap first, then delete the pods of DaemonSet to take effect.
please note that it's unsafe to change the discovery directory of storage class or rename the storage class

**Comment by bcollard** [comment]
Well, let's say I add a new storageClass to the configMap.
Then I delete the DS so that it "restarts" with the correct CM.
I think there will be a problem since the DS template has not changed and the new storageClass "base path" volume will not be mounted on the DS pods, neither under `Volumes:` nor under `volumeMounts:`

**Comment by cofyc** [comment]
you're right, the DS must be updated too when new discovery directories are added

**Comment by bcollard** [comment]
yes and you are forced to use different StorageClass when you want to share the volumes between different applications. Otherwise, the first arrived is the first served, no matter the volume size adequacy, see #150

**Comment by bcollard** [comment]
For what it's worth, here is a step-by-step guide on how I'm managing evolving storage requirements: 
https://www.baptistout.net/posts/dynamize-your-local-storage-kubernetes-ansible/
This uses Kustomize to patch the configmap and the daemonset of the provisioner.

**Comment by cofyc** [comment]
/unassign 
I'm going to unassign myself from this issue. The biggest problem of reloading configuration on the fly is it's not safe to change the configuration dynamically. Manual intervention is often required.
Instead, we should document the upgrade steps in detail.
